x-pack/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -0,0 +1,312 @@
+                selectableMessage: (selectable: boolean) =>
+                  !selectable ? 'Role is reserved' : undefined,
+                onSelectionChange: (selection: Role[]) => this.setState({ selection }),
+              }}
+              pagination={{
+                initialPageSize: 20,
+                pageSizeOptions: [10, 20, 30, 50, 100],
+              }}
+              items={this.getRolesForDisplay()}
+              loading={roles.length === 0}
+              search={{
+                toolsLeft: this.renderToolsLeft(),
+                box: {
+                  incremental: true,
+                },
+                onChange: (query: Record<string, any>) => {
+                  this.setState({
+                    filter: query.queryText,
+                  });
+                },
+              }}
+              sorting={{
+                sort: {
+                  field: 'name',
+                  direction: 'asc',
+                },
+              }}
+              // @ts-ignore missing rowProps typedef
+              rowProps={() => {
+                return {
+                  'data-test-subj': 'roleRow',
+                };
+              }}
+              isSelectable
+            />
+          }
+        </EuiPageContentBody>
+      </EuiPageContent>
+    );
+  };
+
+  private getColumnConfig = (intl: InjectedIntl) => {
+    const reservedRoleDesc = intl.formatMessage({
+      id: 'xpack.security.management.roles.reservedColumnDescription',
+      defaultMessage: 'Reserved roles are built-in and cannot be edited or removed.',
+    });
+
+    return [
+      {
+        field: 'name',
+        name: intl.formatMessage({
+          id: 'xpack.security.management.roles.nameColumnName',
+          defaultMessage: 'Role',
+        }),
+        sortable: true,
+        truncateText: true,
+        render: (name: string, record: Role) => {
+          return (
+            <span>
+              <EuiLink data-test-subj="roleRowName" href={getRoleManagementHref(name)}>
+                {name}
+              </EuiLink>
+              {!isRoleEnabled(record) && (
+                <EuiIconTip
+                  type="alert"
+                  size="xs"
+                  color="warning"
+                  data-test-subj="disabledRoleTip"
+                  title={
+                    <FormattedMessage
+                      id="xpack.security.management.roles.disabledTooltip"
+                      defaultMessage="This role is currently disabled. You may only view or delete it."
+                    />
+                  }
+                />
+              )}
+            </span>
+          );
+        },
+      },
+      {
+        field: 'metadata._reserved',
+        name: (
+          <div>
+            <FormattedMessage
+              id="xpack.security.management.roles.reservedColumnName"
+              defaultMessage="Reserved"
+            />
+            <span className="rolesGridPage__reservedRoleTooltip">
```

Comment:
There's an example on the EUI table docs that shows how to easily apply the tooltip in headers without needing custom classes (which are also duplicated here). But you should also wrap the whole text in the tooltip so they're not relied on just that small icon. So something like:

```jsx
name: (
  <EuiToolTip content={reservedRoleDesc}>
    <span>
      <FormattedMessage ... />
      <EuiIcon size="s" color="subdued" type="questionInCircle" className="eui-alignTop" />
    </span>
  </EuiToolTip>
),
```

Reply:
ah, much better, I'll make that change... thanks!